ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py

```diff
@@ -0,0 +1,340 @@
+        self.ip_addresses = {"Prometheus": "", "Grafana": "", "Redis": ""}
+
+    @j.baseclasses.chatflow_step(title="Network")
+    def network_selection(self):
+        self.network = j.sal.reservation_chatflow.network_select(self, j.me.tid)
+
+    @j.baseclasses.chatflow_step(title="Solution name")
+    def solution_name(self):
+        self.model = j.threebot.packages.tfgrid_solutions.tfgrid_solutions.bcdb_model_get(
+            "tfgrid.solutions.monitoring.1"
+        )
+        self.user_form_data["Solution name"] = j.sal.reservation_chatflow.solution_name_add(self, self.model)
+
+    @j.baseclasses.chatflow_step(title="Access keys")
+    def public_key_get(self):
+        self.user_form_data["Public key"] = self.upload_file(
+            """Please add your public ssh key, this will allow you to access the deployed containers using ssh.
+                    Just upload the file with the key""",
+            required=True,
+        ).split("\n")[0]
+
+    @j.baseclasses.chatflow_step(title="Prometheus container resources")
+    def prometheus_container_resources(self):
+        form = self.new_form()
+        cpu = form.int_ask(
+            "Please add how many CPU cores are needed for the Prometheus container", default=1, required=True
+        )
+        memory = form.int_ask("Please add the amount of memory in MB", default=1024, required=True)
+        rootfs_type = form.single_choice(
+            "Select the storage type for your root filesystem", ["SSD", "HDD"], default="SSD"
+        )
+        rootfs_size = form.int_ask("Choose the amount of storage for your root filesystem in MiB", default=256)
+        form.ask()
+        self.user_form_data["Prometheus CPU"] = cpu.value
+        self.user_form_data["Prometheus Memory"] = memory.value
+        self.user_form_data["Prometheus Root filesystem Type"] = str(rootfs_type.value)
+        self.user_form_data["Prometheus Root filesystem Size"] = rootfs_size.value
+
+        self.prometheus_query["mru"] = math.ceil(self.user_form_data["Prometheus Memory"] / 1024)
+        self.prometheus_query["cru"] = self.user_form_data["Prometheus CPU"]
+        storage_units = math.ceil(self.user_form_data["Prometheus Root filesystem Size"] / 1024)
+        if rootfs_type.value == "SSD":
+            self.prometheus_query["sru"] = storage_units
+        else:
+            self.prometheus_query["hru"] = storage_units
+
+    @j.baseclasses.chatflow_step(title="Prometheus volume details")
+    def prometheus_volume_details(self):
+        form = self.new_form()
+        vol_disk_type = form.drop_down_choice(
+            "Please choose the type of disk for the volume to be attached to the Prometheus container",
+            ["SSD", "HDD"],
+            required=True,
+            default="SSD",
+        )
+        vol_disk_size = form.int_ask("Please specify the volume size", required=True, default=10)
```

Comment:
Add in which unit the size of the volume is given

Reply:
updated here https://github.com/threefoldtech/jumpscaleX_threebot/pull/764/commits/2c57640372d6d533176a56676396ab87810b0f3b#diff-0e24746aa7feb62e0d805614631f5de0R96